cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java

```diff
@@ -184,9 +184,14 @@ private static PublicationDate convertToPublicationDate(LocalDate publishedDate)
     private URI extractHandle() {
         return Optional.ofNullable(cristinObject.getCristinAssociatedUris())
                    .flatMap(CristinMapper::extractArchiveUri)
+                   .map(this::updateHttpScheme)
                    .orElse(null);
     }
 
+    private URI updateHttpScheme(URI uri) {
+        return UriWrapper.fromHost(uri.toString()).addChild(uri.getPath()).getUri();
```

Comment:
why `fromHost(uri.toString())` and not `fromHost(uri.getHost())`?

Reply:
That makes sense